src/Syntax/TreeSitterSyntaxHighlights.re

```diff
@@ -33,8 +33,19 @@ let scopesToStrings = (scopes: list(TreeSitter.Syntax.scope)) => {
   );
 };
 
-let create = (~theme, ~scopeConverter, lines: array(string)) => {
-  let parser = Parser.json();
+let getParserFromScope = language =>
+  switch (language) {
+  | "source.json" => Some(Parser.json())
+  | "source.c" => Some(Parser.c())
+  | "source.cpp" => Some(Parser.cpp())
+  | "source.python" => Some(Parser.python())
+  | "source.js" => Some(Parser.javascript())
+  | "source.ts" => Some(Parser.typescript())
+  | "source.tsx" => Some(Parser.tsx())
+  | _ => None
+  };
+
+let create = (~theme, ~scopeConverter, ~parser, lines: array(string)) => {
```

Comment:
This is inline right now more just for ease of debugging, since I can't link against `reason-tree-sitter` (it complains about inconsistent usage of `editor-core-types`).

This doesn't feel like the place for this to live...but also since the scopes aren't `tree-sitter` scopes (that I can see? They seem to come from the VScode extensions), I don't especially know a nice place for them to live.

It could go into `reason-tree-sitter` and be renamed to something more specific i.e. `getParserFromXScope` where X is whatever this scope is, unless someone has a better idea than me for this.

I originally was going to have it in `reason-tree-sitter` just so the updates for parser could be self-contained, and adding a parser was just a case of adding it to the lib and bumping the version in Oni2....that got ruined when I realised regardless you need to add the corresponding scope converter in this repo anyways.

Reply:
> It could go into reason-tree-sitter and be renamed to something more specific i.e. getParserFromXScope where X is whatever this scope is, unless someone has a better idea than me for this.

I agree - I think, in retrospect, since the only place we can link this is in `esy-tree-sitter`... it makes sense to have them in `reason-tree-sitter`. The nice thing is we could add tests for the particular syntax parsers and scopes.

> that got ruined when I realised regardless you need to add the corresponding scope converter in this repo anyways.

Right! I was thinking... for now... we could also embed the scope converters as just JSON in `reason-tree-sitter`, like:
```
let jsonScopeConverter = {|
{ .. }
|};
```

And then your `getParserFromXScope` function could know about both the actual parser, and the scope converter. Plus we could add tests for individual languages right in `reason-tree-sitter`.

I also don't think this would preclude having dynamically-loaded parsers down the road... we could have a global method in `reason-tree-sitter` that could be like:
```
registerTreeSitter(~scopeName, ~scopeConverterJSON, ~parserDynamicLibraryPath)
```
and just add to our built-in set of tree-sitters...